### PR TITLE
issue #106 - restructuring integration menu dropdown code

### PR DESCRIPTION
**Pull request by bfitzpat**
Signed-off-by: Brian Fitzpatrick <bfitzpat@redhat.com>

### CODE REVIEW

**Comment by bfitzpat** [comment]
@lhein I think this is a little easier to understand (and better yet, it works!) -- but am curious if you see some way to streamline the switch statement. That seems clunky to me even though it works.

**Comment by bfitzpat** [comment]
Once this one is in place, I will work on updating https://github.com/camel-tooling/vscode-camelk/pull/85 and https://github.com/camel-tooling/vscode-camelk/pull/88 and https://github.com/camel-tooling/vscode-camelk/pull/91 and https://github.com/camel-tooling/vscode-camelk/pull/93

**Comment by apupier** [comment]
I think the original problem is fixed by https://github.com/camel-tooling/vscode-camelk/pull/110 , this PR is providing a workaround with some limitations.

This code refactor still makes sense as it is providing cleaner code.
The ignoreFocusOut can be used or not, the good side is that it will avoid a too big same kind of regression , we will just have the up/down arrow not working instead of not being able to use the command at all.

**Comment by bfitzpat** [comment]
Arrows work now without the ignoreFocusOut and it's been rebased on the latest master that includes the fix https://github.com/camel-tooling/vscode-camelk/pull/110. @apupier 

If we get this one approved and merged, I'll work on the other PRs to get them refactored in this manner so they are all a little easier to read.

File: src/IntegrationUtils.ts [hunk 1]
```diff
@@ -28,54 +28,100 @@ const basicIntegration : string = 'Basic - Apache Camel K Integration (no Config
...
 
+const choiceList = [ basicIntegration, 
+	configMapIntegration, 
+	secretIntegration ];
+
 export async function startIntegration(context: vscode.Uri): Promise<any> {
 	return new Promise <any> ( async (resolve, reject) => {
-		await vscode.window.showQuickPick([ basicIntegration, configMapIntegration, secretIntegration ],
-		{
+		const choice : unknown = await vscode.window.showQuickPick(choiceList, {
```
Comment (apupier) [inline src/IntegrationUtils.ts]: shouldn't we use "string | undefined" instead of "unknown"?

Comment (bfitzpat) [reply src/IntegrationUtils.ts]: Considering I had no clue I could do that. :) I just tried it and it worked great. Changing now.

**Comment by bfitzpat** [comment]
Done. Thanks for the help

**Pull request merged**